eventlet/green/http/client.py

```diff
@@ -1450,35 +1450,21 @@ def getresponse(self):
 except ImportError:
     pass
 else:
-    if ssl._has_ssl_wrap_socket:
```

Comment:
hmm... I'm not sure to understand your intentions here.
Your commit message said that we only support py38 ("Simplify given we only support 3.8+") but you removed the sections related to py311 and earlier versions and you keep the code related to py312.

Should we not keep this condition to differentiate Py38-Py311 and Py312+.

Please can you give more details about your thinking.

Reply:
The original comment was wrong. It assumed the new API is new in 3.12, but it actually exists in 3.8 too.

Reply:
ack. thanks for details. Then LGTM